osquery/tables/system/windows/appcompat_shims.cpp

```diff
@@ -0,0 +1,92 @@
+/*
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the BSD-style license found in the
+ *  LICENSE file in the root directory of this source tree. An additional grant
+ *  of patent rights can be found in the PATENTS file in the same directory.
+ *
+ */
+
+#include <string>
+
+#include <osquery/core.h>
+#include <osquery/tables.h>
+
+#include "osquery/tables/system/windows/registry.h"
+
+namespace osquery {
+namespace tables {
+
+struct sdb {
+  std::string description;
+  unsigned long long installTimestamp;
+  std::string path;
+  std::string type;
+};
+
+QueryData genShims(QueryContext& context) {
+  QueryData results;
+  QueryData sdbResults;
+  QueryData shimResults;
+  std::map<std::string, sdb> sdbs;
+
+  queryKey("HKEY_LOCAL_MACHINE",
+           "SOFTWARE\\Microsoft\\Windows "
+           "NT\\CurrentVersion\\AppCompatFlags\\InstalledSDB",
+           sdbResults);
+  for (const auto& rKey : sdbResults) {
+    QueryData appResults;
+    sdb sdb;
+    std::string subkey = rKey.at("subkey");
+    std::string sdbId = subkey.substr(subkey.find("{"), subkey.length());
+    // make sure it's a sane uninstall key
+    queryKey("HKEY_LOCAL_MACHINE", subkey, appResults);
+    for (const auto& aKey : appResults) {
+      if (aKey.at("name") == "DatabaseDescription") {
+        sdb.description = aKey.at("data");
+      }
+      if (aKey.at("name") == "DatabaseInstallTimeStamp") {
+        // take this crazy windows timestamp to a unix timestamp
+        sdb.installTimestamp = std::stoull(aKey.at("data"));
```

Comment:
Yes it looks like filetimeToUnixtime would help, but as you mentioned earlier, it would require turning this into a FILETIME object.